server/controllers/user.js

```diff
@@ -102,8 +146,54 @@ exports.followings = asyncHandler(async (req, res, next) => {
     const username = req.params.username;
 
     const currentUser = await User.findOne({ username: username });
-    res.status(200).json(currentUser.followings);
+
+    // // get all user data
+    const followings = [];
+    for (const f of currentUser.followings) {
+      const t = await User.find({ _id: f }, projection);
+      followings.push(t[0]);
+    }
+
+    res.status(200).json(followings);
+  } catch (err) {
+    res.status(500).json(err)
+  }
+});
+
+// @route GET /users/:username/followSuggestions
+// @desc Get follow Suggestions list
+// @access Private
+exports.followSugestions = asyncHandler(async (req, res, next) => {
+  try {
+    const username = req.params.username;
+    const currentUser = await User.findOne({ username: username });
+
+    // filter users that u already follow
+    const allUsers = await User.find({}, projection);
+
+    const filteredUsers = [];
+
+    for (let user of allUsers) {
+      //check if self
+      if (user._id.equals(currentUser._id)) {
```

Comment:
save `user._id` in a variable since it looks like you're reusing it multiple times